eureka-core/src/test/java/com/netflix/eureka/registry/AwsInstanceRegistryTest.java

```diff
@@ -0,0 +1,100 @@
+package com.netflix.eureka.registry;
+
+import com.netflix.appinfo.AmazonInfo;
+import com.netflix.appinfo.DataCenterInfo;
+import com.netflix.appinfo.InstanceInfo;
+import com.netflix.appinfo.InstanceInfo.InstanceStatus;
+import com.netflix.appinfo.LeaseInfo;
+import com.netflix.discovery.EurekaClient;
+import com.netflix.discovery.EurekaClientConfig;
+import com.netflix.eureka.EurekaServerConfig;
+import com.netflix.eureka.resources.ServerCodecs;
+import org.junit.Test;
+
+
+/**
+ * Created by Nikos Michalakis on 7/14/16.
+ */
+public class AwsInstanceRegistryTest extends InstanceRegistryTest {
+
+    @Test
+    public void testOverridesWithAsgEnabledThenDisabled() {
+        // Regular registration first
+        InstanceInfo myInstance = createLocalUpInstanceWithAsg(LOCAL_REGION_INSTANCE_1_HOSTNAME);
+        registerInstanceLocally(myInstance);
+        verifyLocalInstanceStatus(myInstance.getId(), InstanceStatus.UP);
+
+        // Now we disable the ASG and we should expect OUT_OF_SERVICE status.
+        ((AwsInstanceRegistry) registry).getAwsAsgUtil().setStatus("ASG-YO-HO", false);
```

Comment:
Why not use myInstance.getASGName() here? That way there won't be magic strings.